### PR TITLE
Implement auto publishing workflow and use GitHub actions

### PR DESCRIPTION
**Pull request by remusao**
Hey Sam,

As discussed I have implemented auto-publishing on `@cliqz/url-parser` in this PR. I made the following changes:

1. Make use of GitHub actions (I basically used the same config templates as my other repos).
2. Make use of `auto` to publish on merge and update CHANGELOG automatically.
3. A set of normalized labels will be created and they can be used on PRs to indicate what kind of changes we are making (e.g. performance, bugfix, etc.). Depending on the kind of label, the version will be bumped accordingly (e.g. patch, minor or major).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Bundle
+        run: npm run bundle
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Create release
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (sammacbeth) [inline .github/workflows/release.yml]: These need to be added to the project by an admin right? Or did you already do this?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# top-most EditorConfig file
+root = true
```
Comment (sammacbeth) [inline .editorconfig]: Should this file be committed?

File: .github/workflows/release.yml (hunk 1 above)
Comment (remusao) [reply .github/workflows/release.yml]: I have added `NPM_TOKEN` already (you can see it in the `secrets` section of the repository settings) and `GITHUB_TOKEN` is automatically generated for a short period of time for each GitHub action job so no need to take care of it.

File: .editorconfig (hunk 2 above)
Comment (remusao) [reply .editorconfig]: I think it's nice to have, most editors should be ale to detect it and use its values to adjust parameters of number of spaces for indentation, etc. I can remove it if you'd like but I find it pretty useful (we have it on some other repos like nav-ext).

**Pull request merged**